pkg/kclient/kclient.go

```diff
@@ -63,3 +69,154 @@ func CreateObjectMeta(name, namespace string, labels, annotations map[string]str
 
 	return objectMeta
 }
+
+func (c *Client) CreateTLSSecret(tlsCertificate []byte, tlsPrivKey []byte, componentName string, applicationName string, portNumber int) (*corev1.Secret, error) {
+	// TypeMeta:   metav1.TypeMeta{Kind: "Ingress", APIVersion: "extensions/v1beta1"},
+	// ObjectMeta: metav1.ObjectMeta{Name: i.Labels[urlLabels.URLLabel]},
+	labels := componentlabels.GetLabels(componentName, applicationName, true)
+	portAsString := fmt.Sprintf("%v", portNumber)
+	tlsSecretName := componentName + "-" + portAsString + "-tlssecret"
+	data := make(map[string][]byte)
+	data["tls.crt"] = tlsCertificate
+	data["tls.key"] = tlsPrivKey
+	secretTemplate := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   tlsSecretName,
+			Labels: labels,
+		},
+		Type: corev1.SecretTypeTLS,
+		Data: data,
+	}
+
+	secret, err := c.KubeClient.CoreV1().Secrets(c.Namespace).Create(&secretTemplate)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to create secret %s", tlsSecretName)
+	}
+	return secret, nil
+}
+
+// IngressParamater struct for function createIngress
+type IngressParamater struct {
+	Name          string
+	ServiceName   string
+	IngressDomain string
+	PortNumber    intstr.IntOrString
+	TLSSecretName string
+}
+
+// CreateIngress creates an ingress object for the given service and with the given labels
+// serviceName is the name of the service for the target reference
+// ingressDomain is the ingress domain to use for the ingress
+// portNumber is the target port of the ingress
+func (c *Client) CreateIngress(ingressParam IngressParamater, labels map[string]string) (*extensionsv1.Ingress, error) {
+	fmt.Println("The secret value is " + ingressParam.TLSSecretName)
```

Comment:
please use glog for printing debug messages

Reply:
This message was added when I was doing local testing. Removed in the latest commit.